blockchain/Validation.go

```diff
@@ -2248,19 +2325,476 @@ var (
+			}
+
+			if !cache.HaveCoin(txin.PreviousOutPoint) {
+				if *missingInputs {
+					*missingInputs = true
+				}
+
+				// fMissingInputs and !state.IsInvalid() is used to detect
+				// this condition, don't set state.Invalid()
+				ret = false
+				return
+			}
+		}
+
+		// Are the actual inputs available?
+		if !cache.HaveInputs(*ptx) {
+			ret = state.Invalid(false, model.REJECT_DUPLICATE, "bad-txns-inputs-spent", "")
+			return
+		}
+
+		// Bring the best block into scope.
+		cache.GetBestBlock()
+		valueIn = cache.GetValueIn(ptx)
+
+		// We have all inputs cached now, so switch back to dummy, so we
+		// don't need to keep lock on mempool.
+		cache.Base = &backed
+
+		// Only accept BIP68 sequence locked transactions that can be mined
+		// in the next block; we don't want our mempool filled up with
+		// transactions that can't be mined yet. Must keep pool.cs for this
+		// unless we change CheckSequenceLocks to take a CoinsViewCache
+		// instead of create its own.
+		if !CheckSequenceLocks(ptx, consensus.StandardLocktimeVerifyFlags, &lp, false) {
+			ret = state.Dos(0, false, model.REJECT_NONSTANDARD, "non-BIP68-final", false, "")
+			return
+		}
+	}()
+
+	// Check for non-standard pay-to-script-hash in inputs
+	if GfRequireStandard && !AreInputsStandard(ptx, &cache) {
+		ret = state.Invalid(false, model.REJECT_NONSTANDARD, "bad-txns-nonstandard-inputs", "")
+		return
+	}
+
+	sigOpsCount := GetTransactionSigOpCount(tx, &cache, policy.STANDARD_SCRIPT_VERIFY_FLAGS)
+
+	valueOut := ptx.GetValueOut()
+	fees := int64(valueIn) - valueOut
+	// nModifiedFees includes any fee deltas from PrioritiseTransaction
+	modifiedFees := fees
+	priorityDummy := float64(0)
+	pool.ApplyDeltas(txid, priorityDummy, modifiedFees)
+
+	var inChainInputValue btcutil.Amount
+	priority := cache.GetPriority(ptx, uint32(chainActive.Height()), &inChainInputValue)
+
+	// Keep track of transactions that spend a coinbase, which we re-scan
+	// during reorgs to ensure COINBASE_MATURITY is still met.
+	spendsCoinbase := false
+	for _, txin := range ptx.Ins {
+		coin := cache.AccessCoin(txin.PreviousOutPoint)
+		if coin.IsCoinBase() {
+			spendsCoinbase = true
+			break
+		}
+	}
 
+	entry := mempool.NewTxMempoolEntry(tx, btcutil.Amount(fees), acceptTime, priority, uint(chainActive.Height()),
+		inChainInputValue, spendsCoinbase, int64(sigOpsCount), &lp)
+
+	size := entry.TxSize
+
+	// Check that the transaction doesn't have an excessive number of
+	// sigops, making it impossible to mine. Since the coinbase transaction
+	// itself can contain sigops MAX_STANDARD_TX_SIGOPS is less than
+	// MAX_BLOCK_SIGOPS_PER_MB; we still consider this an invalid rather
+	// than merely non-standard transaction.
+	if uint(sigOpsCount) > policy.MAX_STANDARD_TX_SIGOPS {
+		ret = state.Dos(0, false, model.REJECT_NONSTANDARD, "bad-txns-too-many-sigops",
+			false, strconv.Itoa(sigOpsCount))
+		return
+	}
+
+	var defaultRelaypriority int64
+	if DEFAULT_RELAYPRIORITY {
```

Comment:
convert error

Reply:
this logic convert maybe correct. let us discuss.